src/services/array-functions.js

```diff
@@ -1,48 +1,69 @@
 
 //in the function map, create a new array and store in a variable
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop into the call to fnc
 //add the returned value from fnc to the new array
 //return the new array
 export function map(theArray, fnc){
+  let newArray = [];
+  for(let i = 0; i < array.length; i++){
+    let stuff = fnc(theArray[i]);
+    myArray.push(stuff);
 
+  }
+return myArray;
 }
 
 //create a new array
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
 export function filter(theArray, fnc){
+  let newArray = [];
+  for (let i = 0; i < theArray.length; i++) {
+    console.log(theArray[i]);
+    if (fnc(theArray[i])) {
+      filteredArray.push(theArray[i]);
+    }
+  }
+  return filteredArray;
 
 }
 
 
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
-//fnc will return true or false, if true return the item 
+//fnc will return true or false, if true return the item
 //return null
 export function find(theArray, fnc){
+  let findArray = [];
+  findArray = theArray.find(fnc);
```

Comment:
need to double check the find function. We need to build these functions ourselves from scratch. We shouldn't be using the already built in javascript find function to build our own find function